src/ioctls/sev.rs

```diff
@@ -0,0 +1,680 @@
+#![cfg(feature = "amd-sev")]
+
+use kvm_bindings::*;
+use std::fs::{File, OpenOptions};
+use std::mem::size_of_val;
+use std::os::unix::io::{AsRawFd, RawFd};
+
+use ioctls::{vm::VmFd, Result};
+use kvm_ioctls::*;
+use sev::launch::*;
+use vmm_sys_util::errno;
+use vmm_sys_util::ioctl::ioctl_with_mut_ref;
+
+#[repr(u32)]
+#[allow(dead_code)]
+#[derive(Copy, Clone, Debug)]
+pub enum SevCmd {
+    Init = 0,
+    EsInit,
+
+    LaunchStart,
+    LaunchUpdateData,
+    LaunchUpdateVmsa,
+    LaunchSecret,
+    LaunchMeasure,
+    LaunchFinish,
+
+    SendStart,
+    SendUpdateData,
+    SendUpdateVmsa,
+    SendFinish,
+
+    ReceiveStart,
+    ReceiveUpdateData,
+    ReceiveUpdateVmsa,
+    ReceiveFinish,
+
+    GuestStatus,
+    DebugDecrypt,
+    DebugEncrypt,
+    CertExport,
+}
+
+#[derive(Copy, Clone)]
+pub struct Handle(u32);
+/// Learn from ``SevState`` introduced in QEMU, adding few changes.
+/// An enumeration of SEV state information used during @query-sev.
+///
+/// @Uninit: The guest is uninitialized.
```

Comment:
nit: these should be converted to [doc comments](https://doc.rust-lang.org/stable/rust-by-example/meta/doc.html#doc-comments).